RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc

```diff
@@ -427,17 +429,50 @@ void PulseShapeFitOOTPileupCorrection::fit(int iFit,float &timevalfit,float &cha
    float chargeval2fit = results[3];
    float timeval3fit   = results[4];
    float chargeval3fit = results[5];
+   
+   float deltaT12 = fabs(timeval2fit - timevalfit);
+   float deltaT13 = fabs(timeval3fit - timevalfit);
+   float deltaT23 = fabs(timeval2fit - timeval3fit);
+   
+   // Pick the time closest to zero
+   // Check to see if this pulse overlaps (delta t < 1 ns) with the other pulses
+   // if so do the weighted average for the time and the total energy for energy
 
      if((std::abs(timevalfit)<std::abs(timeval2fit)) && (std::abs(timevalfit)<std::abs(timeval3fit))){
-      timevalfit   = timevalfit;
-      chargevalfit = chargevalfit;
+        if(deltaT12 < overlapLimit){
+          timevalfit = (timeval2fit*chargeval2fit + timevalfit*chargevalfit)/(chargeval2fit+chargevalfit);
+          chargevalfit = chargeval2fit+chargevalfit;     
+        } 
+        if(deltaT13 < overlapLimit) {
+          timevalfit = (timevalfit*chargevalfit + timeval3fit*chargeval3fit)/(chargevalfit+chargeval3fit);
+          chargevalfit = chargevalfit+chargeval3fit;
+        } 
```

Comment:
I think this is ok? If both deltaT12 and deltaT13 satisfy the overlapping pulse criteria, both of the if statements will be used.